cheroot/makefile.py

```diff
@@ -32,8 +34,44 @@ def _flush_unlocked(self):
                 n = self.raw.write(bytes(self._write_buf))
             except io.BlockingIOError as e:
                 n = e.characters_written
+            except OSError as sock_err:
+                error_code = sock_err.errno
+                if error_code in _errors.acceptable_sock_shutdown_error_codes:
+                    # The socket is gone, so just ignore this error.
+                    return
+                raise
+            else:
+                # The 'try' block completed without an exception
+                if n == 0:
```

Comment:
So https://docs.python.org/3/library/io.html#io.RawIOBase.write says

> `None` is returned if the raw stream is set not to block and no single byte could be readily written to it. The caller may release or mutate *b* after this method returns, so the implementation should only access *b* during the method call.

Now should this be handled? With a retry? Also, is the difference between `0` and `None` significant?